spec/factories/courses.rb

```diff
@@ -0,0 +1,33 @@
+FactoryBot.define do
+  factory :course, class: Hash do
+    sequence(:course_code) { |n| "X10#{n}" }
```

Comment:
This could result in course codes which are longer than 4 characters but it shouldn't break anything so it's fine, minor

Reply:
https://github.com/DFE-Digital/manage-courses-backend/blob/7847be1c5e1564f68ed931653f0372c86f2f05a2/db/seeds.rb#L40